src/components/Editor.tsx

```diff
@@ -207,387 +94,44 @@ export const Editor = ({
-      const themeInitialHistory = {
-        history: history,
-        index: history.length - 1,
-      };
-      setThemeInitialHistory(themeInitialHistory);
-      setThemeInitialHistoryFetched(true);
-      return;
-    }
-
-    devLogger.log(
-      "No loadThemeInitialHistory case matched, setting to empty theme."
-    );
-    setThemeInitialHistory({
-      history: [{}],
-      index: 0,
-    });
-    setThemeInitialHistoryFetched(true);
-  }, [
-    setThemeInitialHistory,
-    setThemeInitialHistoryFetched,
-    clearLocalStorage,
-    buildThemeLocalStorageKey,
-  ]);
-
-  // Log PUCK_INITIAL_HISTORY (layout) on load
-  useEffect(() => {
-    if (puckInitialHistory) {
-      devLogger.logData("PUCK_INITIAL_HISTORY", puckInitialHistory);
-    }
-  }, [puckInitialHistory]);
-
-  // Log THEME_INITIAL_HISTORY on load and update theme in editor to reflect save state
-  useEffect(() => {
-    devLogger.logData("THEME_INITIAL_HISTORY", themeInitialHistory);
-    if (themeInitialHistory && themeConfig && templateMetadata?.isThemeMode) {
-      updateThemeInEditor(
-        themeInitialHistory.history[themeInitialHistory.index],
-        themeConfig
-      );
-    }
-  }, [themeInitialHistory, themeConfig, templateMetadata?.isThemeMode]);
-
-  // Handles sending the sendDevThemeSaveStateData on initial load so that nothing isn't rendered for preview.
-  // Subsequent changes are sent via handleChange in ThemeSidebar.tsx.
-  useEffect(() => {
-    if (!themeInitialHistoryFetched || !templateMetadata?.isThemeMode) {
-      return;
-    }
-
-    const themeToSend = themeInitialHistory?.history[themeInitialHistory.index];
-
-    devLogger.logFunc("sendDevThemeSaveStateData useEffect");
-    sendDevThemeSaveStateData({
-      payload: { devThemeSaveStateData: JSON.stringify(themeToSend) },
-    });
-  }, [themeInitialHistoryFetched, themeInitialHistory, templateMetadata]);
-
   const isLoading =
     !puckConfig ||
     !templateMetadata ||
     !document ||
-    !saveStateFetched ||
+    !(layoutSaveStateFetched || templateMetadata?.isThemeMode) ||
     !visualConfigurationDataFetched ||
-    !puckInitialHistoryFetched ||
-    (templateMetadata.isThemeMode &&
-      (!themeDataFetched ||
-        !themeSaveStateFetched ||
-        !themeInitialHistoryFetched));
+    !(themeDataFetched || !templateMetadata?.isThemeMode) ||
+    !(themeSaveStateFetched || !templateMetadata?.isThemeMode);
 
   const progress: number =
-    (100 * // @ts-expect-error adding bools is fine
-      (!!puckConfig +
-        !!templateMetadata +
-        !!document +
-        saveStateFetched +
-        visualConfigurationDataFetched +
-        puckInitialHistoryFetched +
-        (templateMetadata?.isThemeMode // @ts-expect-error adding bools is fine
-          ? themeDataFetched +
-            themeSaveStateFetched +
-            themeInitialHistoryFetched
-          : 0))) /
-    (templateMetadata?.isThemeMode ? 9 : 6);
+    90 * // @ts-expect-error adding bools is fine
```

Comment:
Didn't exactly know what to do with the progress bar because there is still the puckInitialHistoryFetched and/or themeInitialHistoryFetched step in the individual editors before we display Puck but I think those should be near-instant because there's no network requests